### PR TITLE
CI: Add --add-opens in order to support JDK11

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR adds a few ENV variables to the JAVA_OPTS, in order to make ~openjdk11~ oraclejdk11 not output warnings (and illegal reflective access).

<details>

```
 java.lang.reflect.InaccessibleObjectException: Unable to make field private int java.io.FileDescriptor.fd accessible: module java.base does not "opens java.io" to module org.jruby.dist
```

</details>

(I think I will run this a few times.)

See #2147 

---

- enable `oraclejdk11` [Article about the `jdk` setting](https://www.deps.co/guides/travis-ci-latest-java/)
- include all relevant `--add-opens`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -44,9 +44,7 @@ matrix:
 
     # Rails 5.2 builds >= 2.2.2
     - rvm: jruby-head
-      env:
-        - RAILS_VERSION='~> 5.2.0'
-        - JRUBY_OPT=--dev
+      env: RAILS_VERSION='~> 5.2.0' JRUBY_OPT=--dev JAVA_OPTS="--add-opens java.base/sun.nio.ch=org.jruby.dist --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.security=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.security.cert=ALL-UNNAMED --add-opens=java.base/java.util.zip=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.util.regex=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/javax.crypto=ALL-UNNAMED --add-opens=java.management/sun.management=ALL-UNNAMED"
```
Comment (JonRowe) [inline .travis.yml]: Would prefer this over three lines :)

Comment (olleolleolle) [reply .travis.yml]: Right, let's go with that.

**Comment by olleolleolle** [comment]
(Oh, that failed miserably. Somehow, the formatting does not do the right thing.)

**Comment by olleolleolle** [comment]
Haha, without quotes was another disaster:

image

**Comment by olleolleolle** [comment]
@JonRowe Do you know a way to use multi-line & still have it understand the settings?

**Comment by JonRowe** [comment]
The quotes worked to set the env, why that didn't fix Jruby 🤷‍♂

**Comment by olleolleolle** [comment]
OK, thanks! 

I squashed the PR into 1 commit, and that takes me to "Selenium has a tough time loading".


No warnings about `--add-opens`.

We are in the place I wanted to get to.

**Comment by benoittgt** [comment]
Hello @olleolleolle 

I still have breaking build

```
undle binstubs bundler rspec-core rake --force

rm config/application.rb

bin/rake app:template LOCATION='../../example_app_generator/generate_stuff.rb'

rake aborted!

LoadError: load error: childprocess/jruby -- java.lang.reflect.InaccessibleObjectException: Unable to make field private int java.io.FileDescriptor.fd accessible: module java.base does not "opens java.io" to module org.jruby.dist

/home/travis/build/rspec/bundle/jruby/2.5.0/gems/childprocess-1.0.1/lib/childprocess.rb:210:in `<main>'

/home/travis/build/rspec/bundle/jruby/2.5.0/gems/selenium-webdriver-3.142.3/lib/selenium/webdriver.rb:1:in `<main>'

/home/travis/build/rspec/bundle/jruby/2.5.0/gems/selenium-webdriver-3.142.3/lib/selenium/webdriver.rb:20:in `<main>'

/home/travis/build/rspec/bundle/jruby/2.5.0/gems/selenium-webdriver-3.142.3/lib/selenium-webdriver.rb:1:in `block in (root)'

/home/travis/build/rspec/bundle/jruby/2.5.0/gems/selenium-webdriver-3.142.3/lib/selenium-webdriver.rb:20:in `block in <main>'

/home/travis/build/rspec/rspec-rails/tmp/example_app/config/application.rb:7:in `<main>'

/home/travis/build/rspec/rspec-rails/tmp/example_app/Rakefile:1:in `block in (root)'

/home/travis/build/rspec/rspec-rails/tmp/example_app/Rakefile:4:in `<main>'

(See full trace by running task with --trace)

rake aborted!

Command failed with status (1): [bin/rake app:template LOCATION='../../exam...]

/home/travis/build/rspec/rspec-rails/Rakefile:109:in `block in in_example_app'

/home/travis/build/rspec/rspec-rails/Rakefile:107:in `block in in_example_app'

/home/travis/build/rspec/rspec-rails/Rakefile:106:in `in_example_app'

/home/travis/build/rspec/rspec-rails/Rakefile:96:in `block in <main>'

/home/travis/build/rspec/bundle/jruby/2.5.0/gems/rake-12.3.3/exe/rake:27:in `<main>'

Tasks: TOP => smoke:app => generate:stuff

(See full trace by running task with --trace)

The command "script/run_build 2>&1" exited with 1.
```

https://travis-ci.org/rspec/rspec-rails/jobs/557328064

**Comment by JonRowe** [comment]
Can you remind me what this is fixing @olleolleolle?

**Comment by olleolleolle** [comment]
Yes, this makes this JRuby build and run on current JDKs.

- The `--add-opens` directives allow reflection on classes
- With those in place, we can run on the currently-supported JDKs (9 and newer)

If we have older JDKs active, (JDK8, only, at the moment), we can forgo the `--add-opens` directives.

**Pull request merged**